File: 1146-greatest-common-divisor-of-strings/1146-greatest-common-divisor-of-strings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        len1=len(str1)
        len2=len(str2)

        if str1+str2 != str2+str1:
            return ""

        gcdlength= gcd(len1,len2)
        return str1[:gcdlength]
        

File: 1146-greatest-common-divisor-of-strings/1146-greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        
        if(str1+str2 != str2+str1){
            return "";
        }
        int gcdlength = std::gcd(str1.length(),str2.length());
        return str1.substr(0,gcdlength);
        
    }
};